index.d.ts

```diff
@@ -0,0 +1,61 @@
+// Type definitions for gongfu
+// Project: https://github.com/hojberg/gongfu
+// Definitions by: Andrew Goodale <https://github.com/newyankeecodeshop>
+// TypeScript Version: 2.3
+
+import { Component } from "react";
+
+export interface Msg {
+  tag: string;
+}
+
+export type MsgConstructor = (a?: any) => Msg;
+export type Updater = (msg: Msg) => void;
+export type Runner = (updater: Updater) => void;
+export type Setup = (onChange: Updater) => void;
+
+export interface Effect {
+  concat(that: Effect): Effect;
+
+  map(f: (a: any) => any): Effect;
```

Comment:
@hojberg Should the argument to map be of type `Msg`? It looks like it from the code, but the code uses `any` so I'm not sure.

Reply:
Or maybe it should be `map<T extends Msg>()`?

Reply:
@newyankeecodeshop yeah `T extends Msg`, would mean any `Msg` really. That makes sense to me.